subprojects/core/src/main/java/org/gradle/api/internal/AbstractNamedDomainObjectContainer.java

```diff
@@ -64,6 +65,7 @@ public T create(String name, Closure configureClosure) {
 
     public T create(String name, Action<? super T> configureAction) throws InvalidUserDataException {
         assertCanAdd(name);
+        NameValidator.validate(name);
```

Comment:
This is a pretty deep change, that will impact a lot of things, including custom plugins. I don't think we can really claim to know what constitutes a valid name in every domain, and we should perhaps apply this validation only to those thing we know about.